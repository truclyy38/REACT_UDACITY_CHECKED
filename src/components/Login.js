import { useState } from 'react';
import { connect, useSelector } from 'react-redux';
import { Navigate } from 'react-router-dom';
import { login } from '../actions/authen';
import "./Login.css";
import NavBar from './NavBar';

const Login = ({ dispatch }) => {
  const { currentUser, users } = useSelector(state => state)
  const [username, setUsername] = useState("sarahedo");
  // const [password, setPassword] = useState("123456");

  if (!!currentUser) {
    const urlParams = new URLSearchParams(window.location.search);
    const redirectUrl = urlParams.get('url');
    return <Navigate to={redirectUrl} />;
  }

  const submit = (e) => {
    e.preventDefault();
    const user = Object.values(users).find((user) => user.id === username);
    return dispatch(login(user));
  };

  return (<>
    <div className="container">
      <div className="screen">
        <div className="screen__content">
          <form className="login" onSubmit={submit}>
            <h3 className='fw-bold' data-testid="employee-title">Employee Poll</h3>
            <div className="login__field">
              <label htmlFor="username">Username</label>
              <select className='custom-select' data-testid="username" onChange={(e) => setUsername(e.target.value)}>
                {
                  Object.values(users).map((x,i) => {
                    return (
                      <option key={x['id']}>{x['id']}</option>
                    )
                  })
                }
              </select>
            </div>
            <button type="submit" data-testid="submit" className="button login__submit justify-content-center" disabled={!username}>
              <span className="button__text " >Submit</span>
            </button>
          </form>
        </div>
        <div className="screen__background">
          <span className="screen__background__shape screen__background__shape4"></span>
          <span className="screen__background__shape screen__background__shape3"></span>
          <span className="screen__background__shape screen__background__shape2"></span>
          <span className="screen__background__shape screen__background__shape1"></span>
        </div>
      </div>
    </div>
  </>)
}

export default connect()(Login);