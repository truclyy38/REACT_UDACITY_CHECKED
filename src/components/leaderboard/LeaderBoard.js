import { useSelector } from 'react-redux';
import Image from '../shared/Image';

const LeaderBoard = () => {
  const usersState = useSelector(state => state.users);
  const users = usersState ? Object.values(usersState).sort((a, b) => Object.keys(b.answers).length - Object.keys(a.answers).length) : [];
  return (
    <div className=' m-3 p-3'>
      <h5 className='fw-bold'>Leaderboard</h5>
      <table className="table p-3 bg-light table-bordered">
        <thead>
          <tr>
            <th scope="col">#</th>
            <th scope="col">USER</th>
            <th scope="col">ANSWERED</th>
            <th scope="col">CREATED</th>
          </tr>
        </thead>
        <tbody>
          {
            users.map((user, i) => {
              return (
                <tr key={user.id}>
                  <th scope="row">{i + 1}</th>
                  <td>
                    <Image url={user.avatarURL} w={'60px'} />
                    <span className="fw-bold">{user.name}</span>
                  </td>
                  <td>{Object.keys(user.answers).length}</td>
                  <td>{user.questions.length}</td>
                </tr>
              )
            })
          }
        </tbody>
      </table>
    </div >
  )
}

export default LeaderBoard;