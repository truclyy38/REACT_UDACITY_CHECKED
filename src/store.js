import {configureStore} from '@reduxjs/toolkit';
import currentUser from "./reducers/authen";
import users from "./reducers/users";
import questions from "./reducers/questions";

export const store = configureStore({
  reducer: {
    currentUser,
    users,
    questions,
  },
});
