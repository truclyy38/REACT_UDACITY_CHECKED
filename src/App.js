import { Route, Routes } from 'react-router-dom';
import { connect } from "react-redux";
import './App.css';
import Login from './components/Login';
import { useEffect } from 'react';
import { initData } from './actions/initData';
import NavBar from './components/NavBar';
import Dashboard from './components/dashboard/Dashboard';
import ProtectedRoute from './components/shared/ProtectedRoute';
import New from './components/newpolly/New';
import QuestionPol from './components/polly/QuestionPol';
import PageNotFound from './components/error/PageNotFound';
import LeaderBoard from './components/leaderboard/LeaderBoard';


function App({ dispatch }) {
  useEffect(() => {
    dispatch(initData());
  }, [dispatch]);

  return (
    <div className="app">
      <NavBar />
      <Routes>
        <Route path="/*" element={<ProtectedRoute><Dashboard /></ProtectedRoute>} />
        <Route path="/login" exact element={<Login />} />
        <Route path="/leaderboard" element={<ProtectedRoute><LeaderBoard /></ProtectedRoute>} />
        <Route path="/add" element={<ProtectedRoute><New /></ProtectedRoute>} />
        <Route path="/questions/:id" element={<ProtectedRoute><QuestionPol /></ProtectedRoute>} />
        <Route path="/error" element={<PageNotFound />} />
      </Routes>
    </div>
  );
}

export default connect()(App);
