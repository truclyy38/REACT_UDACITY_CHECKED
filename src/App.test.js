import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter, MemoryRouter } from "react-router-dom";

import App from "./App";
import { store } from "./store";

import React from "react";
import { _saveQuestion, _saveQuestionAnswer } from "./_DATA";
import Login from "./components/Login";
import PageNotFound from "./components/error/PageNotFound";
import LeaderBoard from "./components/leaderboard/LeaderBoard";
import New from "./components/newpolly/New";

describe("App Page", () => {
  it("snapshots", () => {
    const view = render(
      <MemoryRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>
    );
    expect(view).toMatchSnapshot();
  });
});

describe("PageNotFound Page", () => {
  it("snapshots", () => {
    const view = render(<PageNotFound />);
    expect(view).toMatchSnapshot();
  });
});

describe("New Page", () => {
  it("snapshots", () => {
    const view = render(
      <MemoryRouter>
        <Provider store={store}>
          <New />
        </Provider>
      </MemoryRouter>
    );
    expect(view).toMatchSnapshot();
  });
});

describe("LeaderBoard Page", () => {
  it("snapshots", () => {
    const view = render(
      <MemoryRouter>
        <Provider store={store}>
          <LeaderBoard />
        </Provider>
      </MemoryRouter>
    );
    expect(view).toMatchSnapshot();
  });
});

describe("function _saveQuestionAnswer", () => {
  it("can save question answer with correct params", async () => {
    const authedUser = "sarahedo";
    const qid = "vthrdm985a262al8qx3do";
    const answer = "optionOne";
    const result = await _saveQuestionAnswer({ authedUser, qid, answer });
    expect(result).toEqual(true);
  });

  it("return error when save question answer with wrong question id", async () => {
    const authUser = "sarahed";
    const questionID = "lalalala";
    const answer = "optionOne";
    try {
      await _saveQuestionAnswer({ authUser, questionID, answer });
    } catch (e) {
      expect(e).toBe("Error");
    }
  });
});

describe("function _saveQuestion", () => {
  it("can save new question with correct author", async () => {
    const optionOneText = "Plan A";
    const optionTwoText = "Plan B";
    const author = "sarahedo";
    const result = await _saveQuestion({
      optionOneText,
      optionTwoText,
      author,
    });
    expect(result).not.toEqual("Error");
  });

  it("Async Function incorrect", async () => {
    const optionOneText = "Plan A";
    const optionTwoText = "Plan B";
    const author = "notexist";
    try {
      await _saveQuestion({ optionOneText, optionTwoText, author });
    } catch (e) {
      expect(e).toBe("Error");
    }
  });
});

describe("Login Page", () => {
  it("should see login form", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );

    const loginTitle = view.queryByTestId("employee-title");
    const username = view.queryByTestId("username");
    const submitButton = view.queryByTestId("submit");

    expect(view).toMatchSnapshot();
    expect(loginTitle).toBeInTheDocument();
    expect(username).toBeInTheDocument();
    expect(submitButton).toBeInTheDocument();
  });

  it("test fireEvent change in login form", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );

    const username = view.queryByTestId("username");

    fireEvent.change(username, { target: { value: "johndoe" } });
    expect(username.value).toBe("johndoe");
  });
});

describe("NewPol", () => {
  it("snapshots", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <New />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  it("should show new pol form", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <New />
        </BrowserRouter>
      </Provider>
    );

    const firstInputLabel = view.queryByTestId("firstOptionLabel");
    const secondInputLabel = view.queryByTestId("secondOptionLabel");
    const submit = view.queryByTestId("submit-new");

    expect(firstInputLabel.textContent).toBe("First Option");
    expect(secondInputLabel.textContent).toBe("Second Option");
    expect(submit.textContent).toBe("Submit");
  });
  it("should be fill with right input", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <New />
        </BrowserRouter>
      </Provider>
    );

    const firstInput = view.queryByTestId("firstOption");
    const secondInput = view.queryByTestId("secondOption");

    fireEvent.change(firstInput, { target: { value: "Option One" } });
    fireEvent.change(secondInput, { target: { value: "Option Two" } });
    expect(firstInput.value).toBe("Option One");
    expect(secondInput.value).toBe("Option Two");
  });
});
