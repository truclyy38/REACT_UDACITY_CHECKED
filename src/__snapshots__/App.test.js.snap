// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Page snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app"
      >
        <div>
          <nav
            class="navbar navbar-expand-lg navbar-light bg-light"
          >
            <div
              class="container-fluid"
            >
              <span
                class="navbar-brand fw-bold"
              >
                Employee Poll App
              </span>
              <button
                aria-controls="navbarSupportedContent"
                aria-expanded="false"
                aria-label="Toggle navigation"
                class="navbar-toggler"
                data-bs-target="#navbarSupportedContent"
                data-bs-toggle="collapse"
                type="button"
              >
                <span
                  class="navbar-toggler-icon"
                />
              </button>
              <div
                class="collapse navbar-collapse"
                id="navbarSupportedContent"
              >
                <ul
                  class="navbar-nav me-auto mb-2 mb-lg-0"
                >
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link active"
                      href="/"
                    >
                      Home
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link active"
                      href="/leaderboard"
                    >
                      Leaderboard
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link active"
                      href="/add"
                    >
                      New
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
        </div>
        <div
          class="container"
        >
          <div
            class="screen"
          >
            <div
              class="screen__content"
            >
              <form
                class="login"
              >
                <h3
                  class="fw-bold"
                  data-testid="employee-title"
                >
                  Employee Poll
                </h3>
                <div
                  class="login__field"
                >
                  <label
                    for="username"
                  >
                    Username
                  </label>
                  <select
                    class="custom-select"
                    data-testid="username"
                  />
                </div>
                <button
                  class="button login__submit justify-content-center"
                  data-testid="submit"
                  type="submit"
                >
                  <span
                    class="button__text "
                  >
                    Submit
                  </span>
                </button>
              </form>
            </div>
            <div
              class="screen__background"
            >
              <span
                class="screen__background__shape screen__background__shape4"
              />
              <span
                class="screen__background__shape screen__background__shape3"
              />
              <span
                class="screen__background__shape screen__background__shape2"
              />
              <span
                class="screen__background__shape screen__background__shape1"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      <div>
        <nav
          class="navbar navbar-expand-lg navbar-light bg-light"
        >
          <div
            class="container-fluid"
          >
            <span
              class="navbar-brand fw-bold"
            >
              Employee Poll App
            </span>
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              class="navbar-toggler"
              data-bs-target="#navbarSupportedContent"
              data-bs-toggle="collapse"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="collapse navbar-collapse"
              id="navbarSupportedContent"
            >
              <ul
                class="navbar-nav me-auto mb-2 mb-lg-0"
              >
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link active"
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link active"
                    href="/leaderboard"
                  >
                    Leaderboard
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link active"
                    href="/add"
                  >
                    New
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </nav>
      </div>
      <div
        class="container"
      >
        <div
          class="screen"
        >
          <div
            class="screen__content"
          >
            <form
              class="login"
            >
              <h3
                class="fw-bold"
                data-testid="employee-title"
              >
                Employee Poll
              </h3>
              <div
                class="login__field"
              >
                <label
                  for="username"
                >
                  Username
                </label>
                <select
                  class="custom-select"
                  data-testid="username"
                />
              </div>
              <button
                class="button login__submit justify-content-center"
                data-testid="submit"
                type="submit"
              >
                <span
                  class="button__text "
                >
                  Submit
                </span>
              </button>
            </form>
          </div>
          <div
            class="screen__background"
          >
            <span
              class="screen__background__shape screen__background__shape4"
            />
            <span
              class="screen__background__shape screen__background__shape3"
            />
            <span
              class="screen__background__shape screen__background__shape2"
            />
            <span
              class="screen__background__shape screen__background__shape1"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LeaderBoard Page snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" m-3 p-3"
      >
        <h5
          class="fw-bold"
        >
          Leaderboard
        </h5>
        <table
          class="table p-3 bg-light table-bordered"
        >
          <thead>
            <tr>
              <th
                scope="col"
              >
                #
              </th>
              <th
                scope="col"
              >
                USER
              </th>
              <th
                scope="col"
              >
                ANSWERED
              </th>
              <th
                scope="col"
              >
                CREATED
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" m-3 p-3"
    >
      <h5
        class="fw-bold"
      >
        Leaderboard
      </h5>
      <table
        class="table p-3 bg-light table-bordered"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              #
            </th>
            <th
              scope="col"
            >
              USER
            </th>
            <th
              scope="col"
            >
              ANSWERED
            </th>
            <th
              scope="col"
            >
              CREATED
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Login Page should see login form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="screen"
        >
          <div
            class="screen__content"
          >
            <form
              class="login"
            >
              <h3
                class="fw-bold"
                data-testid="employee-title"
              >
                Employee Poll
              </h3>
              <div
                class="login__field"
              >
                <label
                  for="username"
                >
                  Username
                </label>
                <select
                  class="custom-select"
                  data-testid="username"
                >
                  <option>
                    sarahedo
                  </option>
                  <option>
                    tylermcginnis
                  </option>
                  <option>
                    johndoe
                  </option>
                </select>
              </div>
              <button
                class="button login__submit justify-content-center"
                data-testid="submit"
                type="submit"
              >
                <span
                  class="button__text "
                >
                  Submit
                </span>
              </button>
            </form>
          </div>
          <div
            class="screen__background"
          >
            <span
              class="screen__background__shape screen__background__shape4"
            />
            <span
              class="screen__background__shape screen__background__shape3"
            />
            <span
              class="screen__background__shape screen__background__shape2"
            />
            <span
              class="screen__background__shape screen__background__shape1"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="screen"
      >
        <div
          class="screen__content"
        >
          <form
            class="login"
          >
            <h3
              class="fw-bold"
              data-testid="employee-title"
            >
              Employee Poll
            </h3>
            <div
              class="login__field"
            >
              <label
                for="username"
              >
                Username
              </label>
              <select
                class="custom-select"
                data-testid="username"
              >
                <option>
                  sarahedo
                </option>
                <option>
                  tylermcginnis
                </option>
                <option>
                  johndoe
                </option>
              </select>
            </div>
            <button
              class="button login__submit justify-content-center"
              data-testid="submit"
              type="submit"
            >
              <span
                class="button__text "
              >
                Submit
              </span>
            </button>
          </form>
        </div>
        <div
          class="screen__background"
        >
          <span
            class="screen__background__shape screen__background__shape4"
          />
          <span
            class="screen__background__shape screen__background__shape3"
          />
          <span
            class="screen__background__shape screen__background__shape2"
          />
          <span
            class="screen__background__shape screen__background__shape1"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`New Page snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex text-center flex-column p-3"
      >
        <h2
          class="card-text fw-bold"
        >
          Would you rather
        </h2>
        <h5
          class="card-title text-muted"
        >
          Create your own poll
        </h5>
        <form>
          <div
            class="input-group mb-3"
          >
            <span
              class="input-group-text"
              data-testid="firstOptionLabel"
            >
              First Option
            </span>
            <input
              aria-describedby="inputGroup-sizing-default"
              aria-label="Sizing example input"
              class="form-control"
              data-testid="firstOption"
              name="firstOption"
              type="text"
              value=""
            />
          </div>
          <div
            class="input-group mb-3"
          >
            <span
              class="input-group-text"
              data-testid="secondOptionLabel"
            >
              Second Option
            </span>
            <input
              aria-describedby="inputGroup-sizing-default"
              aria-label="Sizing example input"
              class="form-control"
              data-testid="secondOption"
              name="firstOption"
              type="text"
              value=""
            />
          </div>
          <div
            class="d-flex text-center justify-content-center"
            style="width: 100%;"
          >
            <button
              class="button login__submit d-flex text-center justify-content-center"
              disabled=""
              style="width: 200px;"
              type="submit"
            >
              <span
                class="button__text"
                data-testid="submit-new"
              >
                Submit
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex text-center flex-column p-3"
    >
      <h2
        class="card-text fw-bold"
      >
        Would you rather
      </h2>
      <h5
        class="card-title text-muted"
      >
        Create your own poll
      </h5>
      <form>
        <div
          class="input-group mb-3"
        >
          <span
            class="input-group-text"
            data-testid="firstOptionLabel"
          >
            First Option
          </span>
          <input
            aria-describedby="inputGroup-sizing-default"
            aria-label="Sizing example input"
            class="form-control"
            data-testid="firstOption"
            name="firstOption"
            type="text"
            value=""
          />
        </div>
        <div
          class="input-group mb-3"
        >
          <span
            class="input-group-text"
            data-testid="secondOptionLabel"
          >
            Second Option
          </span>
          <input
            aria-describedby="inputGroup-sizing-default"
            aria-label="Sizing example input"
            class="form-control"
            data-testid="secondOption"
            name="firstOption"
            type="text"
            value=""
          />
        </div>
        <div
          class="d-flex text-center justify-content-center"
          style="width: 100%;"
        >
          <button
            class="button login__submit d-flex text-center justify-content-center"
            disabled=""
            style="width: 200px;"
            type="submit"
          >
            <span
              class="button__text"
              data-testid="submit-new"
            >
              Submit
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NewPol snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex text-center flex-column p-3"
      >
        <h2
          class="card-text fw-bold"
        >
          Would you rather
        </h2>
        <h5
          class="card-title text-muted"
        >
          Create your own poll
        </h5>
        <form>
          <div
            class="input-group mb-3"
          >
            <span
              class="input-group-text"
              data-testid="firstOptionLabel"
            >
              First Option
            </span>
            <input
              aria-describedby="inputGroup-sizing-default"
              aria-label="Sizing example input"
              class="form-control"
              data-testid="firstOption"
              name="firstOption"
              type="text"
              value=""
            />
          </div>
          <div
            class="input-group mb-3"
          >
            <span
              class="input-group-text"
              data-testid="secondOptionLabel"
            >
              Second Option
            </span>
            <input
              aria-describedby="inputGroup-sizing-default"
              aria-label="Sizing example input"
              class="form-control"
              data-testid="secondOption"
              name="firstOption"
              type="text"
              value=""
            />
          </div>
          <div
            class="d-flex text-center justify-content-center"
            style="width: 100%;"
          >
            <button
              class="button login__submit d-flex text-center justify-content-center"
              disabled=""
              style="width: 200px;"
              type="submit"
            >
              <span
                class="button__text"
                data-testid="submit-new"
              >
                Submit
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex text-center flex-column p-3"
    >
      <h2
        class="card-text fw-bold"
      >
        Would you rather
      </h2>
      <h5
        class="card-title text-muted"
      >
        Create your own poll
      </h5>
      <form>
        <div
          class="input-group mb-3"
        >
          <span
            class="input-group-text"
            data-testid="firstOptionLabel"
          >
            First Option
          </span>
          <input
            aria-describedby="inputGroup-sizing-default"
            aria-label="Sizing example input"
            class="form-control"
            data-testid="firstOption"
            name="firstOption"
            type="text"
            value=""
          />
        </div>
        <div
          class="input-group mb-3"
        >
          <span
            class="input-group-text"
            data-testid="secondOptionLabel"
          >
            Second Option
          </span>
          <input
            aria-describedby="inputGroup-sizing-default"
            aria-label="Sizing example input"
            class="form-control"
            data-testid="secondOption"
            name="firstOption"
            type="text"
            value=""
          />
        </div>
        <div
          class="d-flex text-center justify-content-center"
          style="width: 100%;"
        >
          <button
            class="button login__submit d-flex text-center justify-content-center"
            disabled=""
            style="width: 200px;"
            type="submit"
          >
            <span
              class="button__text"
              data-testid="submit-new"
            >
              Submit
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PageNotFound Page snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Page Not Found
      </h1>
    </div>
  </body>,
  "container": <div>
    <h1>
      Page Not Found
    </h1>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
